client/landing/stepper/declarative-flow/internals/steps-repository/site-vertical/index.tsx

```diff
@@ -33,8 +33,13 @@ const SiteVertical: Step = function SiteVertical( { navigation } ) {
 	const handleSubmit = async ( event: React.FormEvent ) => {
 		event.preventDefault();
 
-		if ( site && vertical ) {
-			const { value, label } = vertical;
+		if ( site ) {
+			let value = '';
+			let label = '';
+			if ( vertical ) {
+				value = vertical.value;
+				label = vertical.label;
```

Comment:
When I go back to the vertical step, clean up the selection and then click the “Continue” button, I found the saved vertical is the same as the previous one.

Reply:
Good catch!